testutil/mock/session_manager_mock.go

```diff
@@ -0,0 +1,123 @@
+package mock
+
+import (
+	"sync"
+
+	"github.com/4chain-ag/go-bsv-middlewares/pkg/middleware/shared"
+)
+
+// SessionManager is a mock implementation of the SessionManager interface.
+type SessionManager struct {
+	mu sync.Mutex
+	// sessions is a map of sessionNonce to a Session
+	sessions map[string]shared.PeerSession
+	// identityKeyToSessions is a map of peerIdentityKey to a list of sessionNonce's
+	identityKeyToSessions map[string][]string
+}
+
+// NewSessionManager creates a new SessionManager.
+func NewSessionManager() *SessionManager {
+	return &SessionManager{
+		sessions:              make(map[string]shared.PeerSession),
+		identityKeyToSessions: make(map[string][]string),
+	}
+}
+
+// AddSession adds a session to the manager, associating it with its sessionNonce and also with its peerIdentityKey.
+func (m *SessionManager) AddSession(session shared.PeerSession) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if session.SessionNonce != nil {
+		m.sessions[*session.SessionNonce] = session
+	}
+
+	if session.PeerIdentityKey != nil {
+		sessionNonces, exists := m.identityKeyToSessions[*session.PeerIdentityKey]
```

Comment:
I would move this logic to separated function (not-exported) and split some ifs. It's really hard to read from that.